src/Controls/src/BindingSourceGen/BindingSourceGeneratorUtilities.cs

```diff
@@ -4,67 +4,39 @@ namespace Microsoft.Maui.Controls.BindingSourceGen;
 
 internal static class BindingGenerationUtilities
 {
-
-    internal static bool IsNullableValueType(ITypeSymbol typeInfo) =>
-        typeInfo is INamedTypeSymbol namedTypeSymbol
-            && namedTypeSymbol.IsGenericType
-            && namedTypeSymbol.ConstructedFrom.SpecialType == SpecialType.System_Nullable_T;
-            
     internal static bool IsTypeNullable(ITypeSymbol typeInfo, bool enabledNullable)
     {
         if (!enabledNullable && typeInfo.IsReferenceType)
         {
             return true;
         }
 
-        if (typeInfo.NullableAnnotation == NullableAnnotation.Annotated)
-        {
-            return true;
-        }
-
-        return IsNullableValueType(typeInfo);
+        return IsNullableValueType(typeInfo) || IsNullableReferenceType(typeInfo);
     }
 
-    internal static TypeDescription CreateTypeNameFromITypeSymbol(ITypeSymbol typeSymbol, bool enabledNullable)
-    {
-        var isNullable = IsTypeNullable(typeSymbol, enabledNullable);
-        return new TypeDescription(
-            GlobalName: GetGlobalName(typeSymbol, isNullable, typeSymbol.IsValueType),
-            IsNullable: isNullable,
-            IsGenericParameter: typeSymbol.Kind == SymbolKind.TypeParameter,
-            IsValueType: typeSymbol.IsValueType);
-    }
+    private static bool IsNullableValueType(ITypeSymbol typeInfo) =>
+        typeInfo is INamedTypeSymbol namedTypeSymbol
+            && namedTypeSymbol.IsGenericType
+            && namedTypeSymbol.ConstructedFrom.SpecialType == SpecialType.System_Nullable_T;
 
-    internal static TypeDescription CreateTypeDescriptionForExplicitCast(ITypeSymbol typeSymbol, bool enabledNullable)
+    private static bool IsNullableReferenceType(ITypeSymbol typeInfo) =>
+        typeInfo.IsReferenceType && typeInfo.NullableAnnotation == NullableAnnotation.Annotated;
+
+    internal static TypeDescription CreateTypeDescriptionFromITypeSymbol(ITypeSymbol typeSymbol, bool enabledNullable)
```

Comment:
Nit: Could this be now just `CreateTypeDescription`? It would make a bunch of lines shorter.